### DISCUSSION TITLE
Setting a `REDIS_PASSWORD` equivalent using environment variables

### DISCUSSION THREAD

**Discussion by binaryfire** [dragonflydb/dragonfly#2182] [Q&A] [answered]
Hi guys

I'm currently using Redis and I'm testing DF as a possible replacement. Is there a way to set the equivalent of `REDIS_PASSWORD` using Docker environment variables?

Thanks

**Comment by chakaz** [comment, marked as answer]
Hey @binaryfire 
Sure, you could set the env var `DFLY_requirepass`, see more [here](https://www.dragonflydb.io/docs/managing-dragonfly/flags)

**Reply by binaryfire** [reply]
@chakaz Great, thanks. Read the docs but missed the bit about DFLY_ prefixing.

**Reply by chakaz** [reply]
Awesome!
Do you mind telling us a bit about your use case, and how you plan to use Dragonfly?

**Reply by binaryfire** [reply]
@chakaz No worries! Trying it for lightweight web app caching.

Just had a look through the flags. I see dbnum defaults to 16; I’m assuming I can set that to 1 if it’s for a single app using only one db? Also are the “io threads” that are set by proactor_threads equivalent to one Redis instance (since Redis is single-threaded)? If so I’ll probably only need 2 or 3.

**Reply by chakaz** [reply]
re/ `dbnum` - don't feel like you need to tune it down, we expect most users to only use 1 db so the overhead should not be noticeable (unless you start using more)
re/ `proactor_threads` - indeed, that's the main difference between Dragonfly and Redis, so feel free to set it to a value of your choice. The default would be to use as many threads as available CPU cores

**Reply by binaryfire** [reply]
Hey @chakaz. One more question- Dragonfly seems like a no-brainer for caching high-traffic apps. But does it still have performance / resource usage benefits over Redis when using a single thread?

When building simple sites, I deploy Redis on the same server as the webapp. These are resource constrained instances so 1 thread and a few hundred mb is all I'd want (or need) for caching. 

I'm guessing that because of your architectural differences (eg. using `io_uring`) the answer is probably yes... But just wanted to check because I can't find any benchmarks for this.

**Reply by romange** [reply]
@binaryfire I do not think that Dragonfly has many advantages of being used as a sidecar together with a small app - I think Redis is actually a good choice for such setups. Dragonfly has been designed for bigger setups spawning dozens of GBS and more and serving requests from a remote server.
